kuksa_databroker/databroker/src/main.rs

```diff
@@ -319,9 +329,35 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
     }
 
-    info!("Listening on {}", addr);
+    let jwt_public_key = match args.get_one::<String>("jwt-public-key") {
+        Some(pub_key_filename) => match std::fs::OpenOptions::new()
+            .read(true)
+            .open(pub_key_filename)
+        {
+            Ok(mut pub_key_file) => {
+                let mut pub_key = String::new();
+                pub_key_file.read_to_string(&mut pub_key).unwrap();
+                info!("Using '{pub_key_filename}' to authenticate access tokens");
+                Some(pub_key)
+            }
+            Err(err) => {
+                error!("Failed to open file {:?}: {}", pub_key_filename, err);
+                None
+            }
+        },
+        None => None,
+    };
 
-    grpc::server::serve_with_shutdown(&addr, broker, shutdown_handler()).await?;
+    match jwt_public_key {
+        Some(pub_key) => {
+            let token_decoder = auth::token_decoder::TokenDecoder::new(pub_key)?;
+            grpc::server::serve_authorized(&addr, broker, token_decoder, shutdown_handler())
+                .await?;
+        }
+        None => {
```

Comment:
I see here a little bit of a problem maybe. If you forget to specify a key file you get the unauthorized version. This could be dangerous for guys that do not read the output from server and then just assume it is working. Maybe it is better to use the insecure flag. Like mentioned above. I'm not entirely sure it's better but just a thought.

Reply:
> If you forget to specify a key file you get the unauthorized version. This could be dangerous for guys that do not read the output from server and then just assume it is working.

I agree. As far as I understand it isn't even possible to disable authorization in the C++ version. That is the more sensible option, which I think should be the default here as well. However, in order to not break existing clients, I think we should wait with making it the default. Perhaps even updating the warning message to that effect (i.e. the warning will turn to an error in the future).

And as I wrote in the comment above.
> `--insecure` specifies whether to allow non-encrypted communication, i.e. not using TLS. It's unrelated to authorization.

Once we add support for TLS, the default should eventually also be to disallow unencrypted communication. But by the same logic, there should be a transition period where it only generates a warning before actually breaking clients by making it the default.

Reply:
One sort-of future proof way might be to include a commandline parameter that describes the access to unauthorized clients, something like

```
--unauthorized_access [ALL,NONE,READ_ONLY,<custom_token>
```

and we could start with "ALL" as default, then sometime later print a warning, that it will change to "NONE" in a later version.

Reply:
Well, is this type of unauthorized access a feature we want to have long term?

Regardless, I think it should be added as a separate feature in that case.

Reply:
1. Add authorization. It is optional and a warning is printed informing the user that authorization will be required (by default) in the future.
2. Add flag to allow unauthorized access (i.e. what you propose). The warning (above) is removed when this flag is used. Clients can start using this flag if they intend to continue allowing unauthorized access (when it becomes required by default).
3. Make authorization required by default.

Reply:
As discussed, this might be a separate feature moving forward. I put the ideas from this discussion here https://github.com/eclipse/kuksa.val/issues/487